File: Management/Models/CanvasModels/Assignments/CanvasAssignmentDate.cs
namespace CanvasModel.Assignments;

public record CanvasAssignmentDate
(
  [property: JsonPropertyName("title")]
  string Title,

  [property: JsonPropertyName("id")]
  ulong? Id = null,

  [property: JsonPropertyName("base")]
  bool? Base = null,

  [property: JsonPropertyName("due_at")]
  DateTime? DueAt = null,

  [property: JsonPropertyName("unlock_at")]
  DateTime? UnlockAt = null,

  [property: JsonPropertyName("lock_at")]
  DateTime? LockAt = null
);

File: nextjs/src/models/canvas/assignments/canvasAssignmentDate.ts
export interface CanvasAssignmentDate {
  title: string;
  id?: number;
  base?: boolean;
  due_at?: string; // ISO 8601 date string
  unlock_at?: string; // ISO 8601 date string
  lock_at?: string; // ISO 8601 date string
}
